<launch>

	<env name="ROSCONSOLE_FORMAT" value="[${severity}] [${walltime}] [${node}]: ${message}" />
	
	<arg name="sensor_name" default="kinect01" />
	<arg name="bag" default="false" />

	<group if="$(arg bag)">
		<node pkg="optar"
					type="nomarker_position_estimator"
					name="nomarker_position_estimator"
					output="screen"
					args="">
					<remap from="arcore_camera" 					to="optar/arcore_camera_bag" />
					<remap from="kinect_camera" 					to="$(arg sensor_name)/hd/image_mono_rect_bag" />
					<remap from="kinect_depth" 						to="$(arg sensor_name)/hd/image_depth_rect_bag" />
					<remap from="kinect_camera_info" 				to="$(arg sensor_name)/hd/camera_info_bag" />
					<remap from="debug_images_topic" 				to="optar/$(arg sensor_name)/debug_images" />
					<remap from="tf"		 						to="tf_bag" /> 
					<remap from="tf_static" 						to="tf_static_bag" /> 
		</node>
	</group>
	<group unless="$(arg bag)">	
		<node pkg="optar"
					type="nomarker_position_estimator"
					name="nomarker_position_estimator"
					output="screen"
					args="">
					<remap from="arcore_camera" 					to="optar/arcore_camera" />
					<remap from="kinect_camera" 					to="$(arg sensor_name)/hd/image_mono_rect" />
					<remap from="kinect_depth" 						to="$(arg sensor_name)/hd/image_depth_rect" />
					<remap from="kinect_camera_info" 				to="$(arg sensor_name)/hd/camera_info" />
					<remap from="debug_images_topic" 				to="optar/$(arg sensor_name)/debug_images" />
		</node>
	</group>
	<node pkg="optar" type="ntp_server"                   name="ntp_server"                   output="screen"/>
	<node pkg="optar" type="skeleton_centroid_publisher" name="skeleton_centroid_publisher" output="screen"/>

</launch>